nextroute/model_sanity_check.go

```diff
@@ -0,0 +1,163 @@
+package nextroute
+
+import (
+	"time"
+
+	"github.com/nextmv-io/sdk/connect"
+)
+
+// SanityCheckDepth is the depth of the sanity check.
+type SanityCheckDepth int
+
+const (
+	// SanityCheckLow checks if there is at least one move per plan unit.
+	SanityCheckLow SanityCheckDepth = iota
+	// SanityCheckMedium checks same as SanityCheckLow.
+	SanityCheckMedium
+	// SanityCheckHigh count for each plan unit if it fits on a vehicle and
+	// therefor how many units fit on a vehicle.
+	SanityCheckHigh
+)
+
+// ModelSanityCheck is the interface for the sanity check.
+type ModelSanityCheck interface {
+	// Check runs the sanity check. It returns the sanity check output and an
+	// error if the sanity check itself fails. Duration is the maximum time
+	// allowed for the sanity check to run. If the sanity check takes longer
+	// the remark will be "timeout". Depth is the depth of the sanity check.
+	Check(duration time.Duration, depth SanityCheckDepth) (SanityCheckOutput, error)
+}
+
+// SanityCheckOutput is the output of the sanity check.
+type SanityCheckOutput struct {
+	// Remark is the remark of the sanity check. It can be "ok", "timeout" or
+	// anything else that should explain itself.
+	Remark string `json:"remark"`
+	// Error is the error raised during the sanity check. It can be empty.
+	Error string `json:"error"`
+	// InitialSolution is the sanity check of the initial solution.
+	InitialSolution SanityCheckInitialSolution `json:"initial_solution"`
+	// PlanUnits is the sanity check of the plan units.
+	PlanUnits SanityCheckPlanUnits `json:"plan_units"`
+	// Vehicles is the sanity check of the vehicles.
+	Vehicles SanityCheckVehicles `json:"vehicles"`
+}
+
+// SanityCheckInitialSolution is the sanity check of the initial solution. The
+// initial solution is the solution created by the solver with the initial stops
+// that are fixed. The non-fixed stops are not planned.
+type SanityCheckInitialSolution struct {
+	// Feasible is true if the initial solution is feasible.
+	Feasible bool `json:"feasible"`
+	// DurationToCreate is the duration to create the initial solution.
+	DurationToCreate float64 `json:"duration_to_create"`
```

Comment:
Is this really the performance metric? I.e., the time it took to create the initial solution? Maybe we could add the unit to the comment here (I suppose it's seconds).